test/video_room_test.exs

```diff
@@ -1,48 +1,238 @@
 defmodule Janus.Plugin.VideoRoomTest do
   use ExUnit.Case
   import Mock
+  import VideoRoomTest.Helper
   alias Janus.Plugin.VideoRoom
+  alias Janus.Plugin.VideoRoom.Errors
 
   @id 1
   @room_name "room_name"
+  @handle_id 1
 
-  describe "create_room sends create room request through connection" do
-    test "returning ok tuple with room id when call succeeds" do
+  describe "create_room/6 sends create room request through connection and" do
+    test "returns ok tuple with room id on success" do
       description = "A room description"
 
-      with_mock Janus.Connection,
-        call: fn _, message ->
+      with_mock Janus.Session,
+        execute_request: fn _, message ->
           assert message[:body][:room] == @room_name
           assert message[:body][:description] == description
           {:ok, %{"videoroom" => "created", "room" => @id}}
         end do
         room_props = %VideoRoom{description: description}
 
         assert {:ok, @id} ==
-                 VideoRoom.create_room(@room_name, room_props, Janus.Conenction, 1, 1)
+                 VideoRoom.create_room(Janus.Session, @room_name, room_props, nil, nil)
       end
     end
 
-    test "returning an error when room already_exists" do
-      with_mock Janus.Connection,
-        call: fn _, _message ->
-          {:ok, %{"error_code" => 427, "videoroom" => "event"}}
+    test "returns an error when room already_exists" do
+      code = Errors.code(:room_already_exists)
+
+      with_mock Janus.Session,
+        execute_request: fn _, _message ->
+          {:ok, error_message(code)}
+        end do
+        assert {:error, Errors.error(code)} ==
+                 VideoRoom.create_room(Janus.Session, @room_name, %VideoRoom{}, nil, nil)
+      end
+    end
+
+    test_connection_error(&VideoRoom.create_room/5, [
```

Comment:
Have you tested how this test looks when it fails?

Reply:
Yes, it looks alright. 
```elixir
 8) test edit/5 sends edit room request through connection and returns an error on connection error (Janus.Plugin.VideoRoomTest)
     test/video_room_test.exs:72
     match (=) failed
     code:  assert :ok = apply(&VideoRoom.edit/5, [Janus.Session, @room_name, %VideoRoom{}, @handle_id, nil])
     left:  :ok
     right: {:error, :important_error}
     stacktrace:
       test/video_room_test.exs:72: (test)
```

Reply:
decent enough